src/libraries/System.Text.Json/Common/JsonIgnoreCondition.cs

```diff
@@ -10,12 +10,7 @@ namespace System.Text.Json.Serialization
     /// a property or field is ignored during serialization and deserialization. This option
     /// overrides the setting on <see cref="JsonSerializerOptions.DefaultIgnoreCondition"/>.
     /// </summary>
-#if BUILDING_SOURCE_GENERATOR
-    internal
-#else
-    public
-#endif
-    enum JsonIgnoreCondition
```

Comment:
Why is this changing?

Reply:
If it's necessary for unit testing could we leave them as internal and use InternalsVisibleTo from the unit tests?  It's probably not a huge deal, but exposing public surface area from the source generator does create an opportunity for someone to reference it and depend on these.  We intend to not support that, but... yeah.

Reply:
I discussed this with @ericstj and @layomia and we figured that these wouldn't be public in practice since there's no easy way for users to reference them directly. The source generator classes themselves are public already but we don't particularly care about keeping them backward compatible. The SDK assemblies appear to also have public types for the same reason. I'd rather we fall back to IVT if there's strong pushback to the idea of making the model types public.

cc @ViktorHofer

Reply:
I don't have a strong preference here just wanted to share that I noticed some issues with it. I.e. winforms heavily depends on IVT for their testing and when adding polyfills to netstandard2.0 like NullableAttributes.cs ([which runtime does implicitly](https://github.com/dotnet/runtime/blob/93a811f1ef273e64bd2a2280bce50dfdd8be6fc9/src/libraries/Directory.Build.targets#L165-L171) you get compiler errors. That's because the internal types are visible by the consumer which clash with the ones provided via the framework, i.e. when targeting .NETCoreApp.

Unsure if our source generator tests reference the source generator output assemblies here though.